easybuild/easyblocks/s/siesta.py

```diff
@@ -0,0 +1,342 @@
+                'Eig2DOS/Eig2DOS',
+                'Gen-basis/ioncat', 'Gen-basis/gen-basis',
+                'Grid/cdf2grid', 'Grid/cdf_laplacian', 'Grid/cdf2xsf',
+                'Grid/grid2cube',
+                'Grid/grid_rotate', 'Grid/g2c_ng', 'Grid/grid2cdf', 'Grid/grid2val',
+                'Helpers/get_chem_labels',
+                'HSX/hs2hsx', 'HSX/hsx2hs',
+                'JobList/Src/getResults', 'JobList/Src/countJobs',
+                'JobList/Src/runJobs', 'JobList/Src/horizontal',
+                'Macroave/Src/macroave',
+                'ON/lwf2cdf',
+                'Optimizer/simplex', 'Optimizer/swarm',
+                'pdosxml/pdosxml',
+                'Projections/orbmol_proj',
+                'SiestaSubroutine/FmixMD/Src/driver',
+                'SiestaSubroutine/FmixMD/Src/para',
+                'SiestaSubroutine/FmixMD/Src/simple',
+                'STM/simple-stm/plstm', 'STM/ol-stm/Src/stm',
+                'VCA/mixps', 'VCA/fractional',
+                'Vibra/Src/vibra', 'Vibra/Src/fcbuild',
+                'WFS/info_wfsx', 'WFS/wfsx2wfs',
+                'WFS/readwfx', 'WFS/wfsnc2wfsx', 'WFS/readwf', 'WFS/wfs2wfsx',
+            ]
+
+            if LooseVersion(self.version) <= LooseVersion("4.0"):
+                expected_utils.extend([
+                    'Bands/new.gnubands',
+                    'TBTrans/tbtrans',
+                ])
+
+            if LooseVersion(self.version) >= LooseVersion("4.0"):
+                expected_utils.extend([
+                    'SiestaSubroutine/ProtoNEB/Src/protoNEB',
+                    'SiestaSubroutine/SimpleTest/Src/simple_pipes_parallel',
+                    'SiestaSubroutine/SimpleTest/Src/simple_pipes_serial',
+                    'Sockets/f2fmaster', 'Sockets/f2fslave',
+                ])
+
+            if LooseVersion(self.version) >= LooseVersion("4.1"):
+                expected_utils.extend([
+                    'Bands/gnubands',
+                    'Grimme/fdf2grimme',
+                    'SpPivot/pvtsp',
+                    'TS/ts2ts/ts2ts', 'TS/tshs2tshs/tshs2tshs', 'TS/TBtrans/tbtrans',
+                ])
+
+            for f in expected_utils:
+                shutil.copy(os.path.join(self.cfg['start_dir'], 'Util', f), bindir)
+
+        if self.cfg['with_transiesta']:
+            # Build transiesta
+            try:
+                os.chdir(obj_dir)
+            except OSError, err:
+                raise EasyBuildError("Failed to change back to Obj dir: %s", err)
+
+            run_cmd('make clean', log_all=True, simple=True, log_output=True)
+            run_cmd('make %s transiesta' % par, log_all=True, simple=True, log_output=True)
+
+            shutil.copy(os.path.join(self.cfg['start_dir'], 'Obj', 'transiesta'),
+                        bindir)
+
+    def build_step(self):
+        """No build step for Siesta."""
+        pass
+
+    def install_step(self):
+        """Custom install procedure for Siesta."""
+
+        try:
```

Comment:
no need for the `try` anymore if you use `copy_*` functions, they do proper error reporting already